scripts/dts/new_extract.py

```diff
@@ -81,6 +81,27 @@ def main():
                 write_regs(dev, out)
                 write_aliases(dev, out)
 
+                if dev.bus:
+                    bus_type = "BUS_{}".format(str2ident(dev.bus))
+                    define = "DT_{}_{}".format(str2ident(dev.matching_compat), bus_type)
+                    print("#define {}\t1".format(define), file=out)
+                    dev_aliases = dev_path_aliases(dev, bus_type) + dev_instance_aliases(dev, bus_type)
+                    for alias in dev_aliases:
+                        # Avoid writing aliases that overlap with the base identifier for
+                        # the register
+                        if alias != define:
+                            print("#define {}\t{}".format(alias, define), file=out)
+
+                    bus_name = str2ident(dev.parent._node.props['label'].to_string())
+                    define = "{}_BUS_NAME".format(dev_ident(dev))
+                    print('#define {}\t"{}"'.format(define, bus_name), file=out)
+                    dev_aliases = dev_path_aliases(dev, "BUS_NAME") + dev_instance_aliases(dev, "BUS_NAME")
+                    for alias in dev_aliases:
+                        # Avoid writing aliases that overlap with the base identifier for
+                        # the register
+                        if alias != define:
+                            print("#define {}\t{}".format(alias, define), file=out)
```

Comment:
`main()` is getting a bit long. Could you move this out into a separate function?